warehouse/ingest-core/src/main/java/datawave/ingest/mapreduce/job/ShardedTableMapFile.java

```diff
@@ -349,7 +350,19 @@ public static Path createShardedMapFile(Logger log, Configuration conf, Path wor
                     Range range = new Range();
                     Locations locations = tableOps.locate(shardedTableName, Collections.singletonList(range));
                     List<TabletId> tabletIds = locations.groupByRange().get(range);
-                    tabletIds.forEach(tId -> splitToLocation.put(tId.getEndRow(), locations.getTabletLocation(tId)));
+                    try {
+                        if (tabletIds.size() > 1)
+                            tabletIds.removeIf(tabletId -> tabletId.getEndRow() == null); // throws UnsupportedOperationException always. This should work
+                    } catch (UnsupportedOperationException use) {
+                        log.warn(use.getMessage() + " could not filter null end rows with the removeIf lambda function.");
+                    }
+                    
+                    for (TabletId tabletId : tabletIds)
+                        if (tabletId.getEndRow() != null)
+                            splitToLocation.put(tabletId.getEndRow(), locations.getTabletLocation(tabletId));
+                    // Would like to use line 365 but it does not filter out the null endRow tabletId
+                    // and line 355 with tabletIds.removeIf(tabletId -> tabletId.getEndRow() == null) fails to do it.
+                    // tabletIds.forEach(tId -> splitToLocation.put(tId.getEndRow(), locations.getTabletLocation(tId)));
```

Comment:
Why not something like:
```
// @formatter:off
tabletIds.stream()
    .filter(tId -> tId.getEndRow() != null)
    .forEach(tId -> splitToLocation.put(tId.getEndRow(), locations.getTabletLocation(tId)));
// @formatter:on
```

Reply:
That is the way to go.  I am still getting comfortable with the stream API.  I will make the change tomorrow and make a new pull request.

Reply:
I won't be around an IDE for a few days so I'm not exactly sure how this function is being used. If we drop the tablet with a null end key, then we'll never return the location of the final tablet for each table. If TreeMap won't work with the overridden comparator then maybe we need a new data structure.

Reply:
@jschmidt10, have a look at the [original PR](https://github.com/NationalSecurityAgency/datawave/pull/321) that caused this bug. Previously we were using some Accumulo internal APIs to get the locations.  If you look at the [convertToTextStringMap](https://github.com/NationalSecurityAgency/datawave/pull/321/files#diff-955e637d394bf9ae192422a56c43fd37L430) function that we passed the results through, it already filtered out locations with a null end row. One could argue that was always a bug although for our sharded tables where we always expect things to be pre-split, we never use the last tablet with the null end row.

Reply:
Ah, fair enough. Thanks!